ngraph/test/type_prop/adaptive_avg_pool.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/type_prop.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(type_prop, adaptive_avg_pool)
+{
+    const PartialShape arg_shape{1, 6, 8, 9};
+    const vector<int64_t> output_shape{5, 7};
+
+    auto data = make_shared<op::Parameter>(element::f32, arg_shape);
+    auto out_shape = op::Constant::create<int64_t>(element::i64, Shape{2}, output_shape);
+    auto adaptive_pool = make_shared<op::v8::AdaptiveAvgPool>(data, out_shape);
+
+    ASSERT_TRUE(adaptive_pool->get_output_partial_shape(0).same_scheme({1, 6, 5, 7}));
+}
+
+TEST(type_prop, adaptive_avg_pool_dyn_batch)
+{
+    const PartialShape arg_shape{Dimension::dynamic(), 6, 8, 9};
```

Comment:
Maybe test with dynamic rank can be added?
Something like:
const PartialShape arg_shape = PartialShape::dynamic()?